server/src/main/java/org/elasticsearch/search/aggregations/support/ValuesSourceType.java

```diff
@@ -92,4 +92,8 @@ default DocValueFormat getFormatter(String format, ZoneId tz) {
      * @return the name of the Values Source Type
      */
     String typeName();
+
+    default RuntimeException getUnmappedException(ValuesSourceConfig config, String name) {
```

Comment:
I don't love passing a config in here.  I get why, but I'd almost rather we passed in a string and let the caller resolve this.  This is just weirdly recursive, since the config is (generally) the thing that holds the VST.

Maybe this takes two strings, and we add a method to ValuesSourceConfig that calls it with the description? IDK.

Also, I know I suggested the name `getUnmappedException` earlier, but seeing it out of context on `ValuesSourceType` I realized that we use "unmapped" to mean something else generally, and this might be confusing.  How do you feel about `getUnregisteredException` instead?  And either way, this method needs some javadoc.